controllers/seedgen_controller.go

```diff
@@ -0,0 +1,630 @@
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	seedgenv1alpha1 "github.com/openshift-kni/lifecycle-agent/api/seedgenerator/v1alpha1"
+	"github.com/openshift-kni/lifecycle-agent/controllers/utils"
+
+	"github.com/openshift-kni/lifecycle-agent/ibu-imager/clusterinfo"
+	"github.com/openshift-kni/lifecycle-agent/ibu-imager/ops"
+	"github.com/openshift-kni/lifecycle-agent/internal/common"
+
+	commonUtils "github.com/openshift-kni/lifecycle-agent/utils"
+)
+
+// SeedGeneratorReconciler reconciles a SeedGenerator object
+type SeedGeneratorReconciler struct {
+	client.Client
+	Log      logr.Logger
+	Scheme   *runtime.Scheme
+	Recorder record.EventRecorder
+	Executor ops.Execute
+}
+
+var (
+	clusterName            string
+	lcaImage               string
+	seedgenWorkingDir      = "/tmp/ibu-seedgen-orch"
+	seedgenAuthFile        = filepath.Join(seedgenWorkingDir, "auth.json")
+	storedSeedgenCR        = filepath.Join(seedgenWorkingDir, "seedgen-cr.json")
+	storedSeedgenSecret    = filepath.Join(seedgenWorkingDir, "seedgen-secret.json")
+	storedManagedClusterCR = filepath.Join(seedgenWorkingDir, "managedcluster.json")
+)
+
+//+kubebuilder:rbac:groups=lca.openshift.io,resources=seedgenerators,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=lca.openshift.io,resources=seedgenerators/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=lca.openshift.io,resources=seedgenerators/finalizers,verbs=update
+//+kubebuilder:rbac:groups="",resources=configmaps,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups="",resources=events,verbs=create;patch
+//+kubebuilder:rbac:groups="",resources=secrets,verbs=get;list;watch;delete
+//+kubebuilder:rbac:groups="",resources=namespaces,verbs=get;list;watch;delete
+//+kubebuilder:rbac:groups="",resources=pods,verbs=get;list;watch
+//+kubebuilder:rbac:groups=config.openshift.io,resources=clusterversions,verbs=get;list;watch
+//+kubebuilder:rbac:groups=rbac.authorization.k8s.io,resources=clusterroles,verbs=delete
+//+kubebuilder:rbac:groups=rbac.authorization.k8s.io,resources=clusterrolebindings,verbs=delete
+//+kubebuilder:rbac:groups=apiextensions.k8s.io,resources=customresourcedefinitions,verbs=delete
+//+kubebuilder:rbac:groups=cluster.open-cluster-management.io,resources=managedclusters,verbs=get;list;watch;delete
+
+// Create an API client for hub requests (ACM)
+func (r *SeedGeneratorReconciler) createHubClient(hubKubeconfig []byte) (hubClient client.Client, err error) {
+	config, err := clientcmd.RESTConfigFromKubeConfig(hubKubeconfig)
+	if err != nil {
+		err = fmt.Errorf("Failed RESTConfigFromKubeConfig: %w", err)
+		return
+	}
+
+	hubClient, err = client.New(config, client.Options{Scheme: r.Scheme})
+	if err != nil {
+		err = fmt.Errorf("Failed to create hub client: %w", err)
```

Comment:
s/Failed/failed

there are many across this file

Reply:
Why would you want this lowercase?

Reply:
Error string should not be capitalized or end with punctuation mark 
 Inspection info: Reports format issues in error strings.

Example:
err := fmt.Errorf("Cannot read the file!")
log.Printf("Reading %s: %v", file, err)

According to [Error Strings at github.com/golang](https://github.com/golang/go/wiki/CodeReviewComments#error-strings), error strings should not be capitalized or end with punctuation because they might appear among other context.

Reply:
Interesting. Maybe there's a golangci-lint check we can turn on for that

Reply:
I was about to say the same thing, as it does happen for TALM. I thought we copied the linters and config from TALM ...

Reply:
The .golangci.yml is the same for both repos

Reply:
There are a couple of related golangci-lint plugins I'd like to enable, and a config I can do to add a check for the capitalized Errorf, but I'd like to do that as a separate PR. In the meantime, I can fix the capitals in this code, and create a separate PR to address the other issues and add linters

Reply:
https://github.com/openshift-kni/lifecycle-agent/pull/130